pennylane/ops/op_math/pow_class.py

```diff
@@ -0,0 +1,285 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines the symbolic operation that indicates the power of an operator.
+"""
+from copy import copy
+from scipy.linalg import fractional_matrix_power
+
+from pennylane.operation import (
+    DecompositionUndefinedError,
+    SparseMatrixUndefinedError,
+    PowUndefinedError,
+    Operator,
+    Operation,
+    Observable,
+)
+from pennylane.queuing import QueuingContext, apply
+
+from pennylane import math as qmlmath
+
+_superscript = str.maketrans("0123456789.+-", "⁰¹²³⁴⁵⁶⁷⁸⁹⋅⁺⁻")
+
+
+# pylint: disable=no-member
+class PowOperation(Operation):
+    """Operation-specific methods and properties for the ``Pow`` class.
```

Comment:
What is the overall function of this mix-in class right now? Is it a conceptual piece, for now, that would be extended in the future?

Reply:
It will be reduced in the future.  We want to start eliminating the distinctions between `Operator`, `Operation`, and `Observable`.  Part of that will be getting rid of in-place inversion, and this PR will allow us to start the process.